tests/anoph/conftest.py

```diff
@@ -584,6 +584,318 @@ def simulate_aim_variants(path, contigs, snp_sites, n_sites_low, n_sites_high):
+    #     - RawCov [2D array] [int] [-1 to 18465+ for n_sites for for n_samples]
+    #   - samples [1D array] [str for n_samples]
+    #   - variants
+    #      - END [1D array] [int for n_snp_sites]
+    #      - POS [1D array] [int for n_snp_sites]
+    # - sample_coverage_variance [1D array] [float] [0 to 0.5 for n_samples]
+    # - sample_is_high_variance [1D array] [bool] [True or False for n_samples]
+    # - samples [1D array] [str]
+
+    # Get a random probability for a sample being high variance, between 0 and 1.
+    p_variance = np.random.random()
+
+    # Open a zarr at the specified path.
+    root = zarr.open(zarr_path, mode="w")
+
+    # Create samples array.
+    df_samples = pd.read_csv(metadata_path)
+    samples = df_samples["sample_id"].values
+    root.create_dataset(name="samples", data=samples, dtype=str)
+
+    # Get the number of samples.
+    n_samples = len(df_samples)
+
+    # Simulate sample_coverage_variance array.
+    sample_coverage_variance = np.random.uniform(low=0, high=0.5, size=n_samples)
+    root.create_dataset(name="sample_coverage_variance", data=sample_coverage_variance)
+
+    # Simulate sample_is_high_variance array.
+    sample_is_high_variance = np.random.choice(
+        [False, True], size=n_samples, p=[1 - p_variance, p_variance]
+    )
+    root.create_dataset(name="sample_is_high_variance", data=sample_is_high_variance)
+
+    for contig in contigs:
+        # Create the contig group.
+        contig_grp = root.require_group(contig)
+
+        # Create the calldata group for this contig.
+        calldata_grp = contig_grp.require_group("calldata")
+
+        # Get the SNP positions for this contig.
+        snp_pos = snp_sites[f"{contig}/variants/POS"][:]
+
+        # Get the number of sites (SNP positions) for this contig.
+        n_sites = snp_pos.shape[0]
+
+        # Simulate CN, NormCov, RawCov under calldata.
+        cn = np.random.randint(low=-1, high=12, size=(n_sites, n_samples))
+        normCov = np.random.randint(low=0, high=356, size=(n_sites, n_samples))
+        rawCov = np.random.randint(low=-1, high=18465, size=(n_sites, n_samples))
+        calldata_grp.create_dataset(name="CN", data=cn)
+        calldata_grp.create_dataset(name="NormCov", data=normCov)
+        calldata_grp.create_dataset(name="RawCov", data=rawCov)
+
+        # Create the samples dataset (again) for this contig.
+        contig_grp.create_dataset(name="samples", data=samples, dtype=str)
+
+        # Create variants group for this contig.
+        variants_grp = contig_grp.require_group("variants")
+
+        # Simulate POS under variants.
+        # Note: choosing a random subset would cause dimension size conflicts.
+        variants_grp.create_dataset(name="POS", data=snp_pos)
+
+        # Simulate END under variants.
+        random_multipliers = np.random.randint(2, 12, size=len(snp_pos))
+        end = random_multipliers * snp_pos
+        variants_grp.create_dataset(name="END", data=end)
```

Comment:
See comment above, POS should be window starts (1, 301, 601, ...) and END should be window ENDs (300, 600, 900, ...).

Reply:
Thanks. Have done.